tests/macros/addonsidebar.test.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * @prettier
+ */
+const url = require('url');
+
+const { JSDOM } = require('jsdom');
+
+const { beforeEachMacro,
+        describeMacro,
+        itMacro,
+        lintHTML } = require('./utils');
+
+const SUMMARIES = {
+    'en-US': new Set([
+        'Getting started',
+        'Concepts',
+        'User interface',
+        'How to',
+        'Porting',
+        'Firefox workflow',
+        'JavaScript APIs',
+        'Manifest keys',
+        'Browser themes',
+        'Publishing add-ons',
+        'Distributing add-ons',
+        'Channels'
+    ]),
+    'fr': new Set([
+        'Démarrage',
+        'Concepts',
+        'Interface Utilisateur',
+        'Mode d\'emploi',
+        'Portage',
+        'Déroulement avec Firefox',
+        'Les API JavaScript',
+        'Clés de manifeste',
+        'Thème de navigateur',
+        'Publication de votre extension',
+        'Distribuer votre module',
+        'Canaux de discussions'
+    ]),
+    'ja': new Set([
+        '始めましょう',
+        '概念',
+        'ユーザーインターフェイス',
+        '逆引きリファレンス',
+        '移行',
+        'Firefox でのワークフロー',
+        'JavaScript API 群',
+        'Manifest keys',
+        'ブラウザのテーマ',
+        'アドオンを公開する',
+        'アドオンの配布',
+        'チャンネル'
+    ]),
+};
+
+const MANIFEST_SLUG = 'Mozilla/Add-ons/WebExtensions/manifest.json';
+
+function getMockResultForFetchJSONResource(doc_url) {
+    const locale = url.parse(doc_url).pathname.split('/')[1];
+    return {
+        locale: `${locale}`,
+        url: `/${locale}/docs/${MANIFEST_SLUG}`,
+        subpages: [
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/author`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/author`,
+                title: 'author'
+            },
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/background`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/background`,
+                title: 'background'
+            },
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/theme`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/theme`,
+                title: 'theme'
+            },
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/version`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/version`,
+                title: 'version'
+            },
+        ],
+        slug: MANIFEST_SLUG,
+        title: 'manifest.json'
+    };
+}
+
+function checkSidebarDom(html, locale, is_under_web_ext_api=false) {
```

Comment:
You're outing yourself as a Python programmer with all the underscores. :-)

I don't object to that style, and given that this is a test, I don't think it  is something you need to change.

Reply:
That's for sure! 😃 I honestly didn't even think twice about this, and now I see how incongruent it is! I'll change this to camel case.